From 544ea3d725a7438c2f6b605551016b583190f9cb Mon Sep 17 00:00:00 2001
From: "hainan.wang" <hainan.wang@unisoc.com>
Date: Thu, 27 Oct 2022 10:01:26 +0800
Subject: [PATCH] Bug #2057864-GLP2.0-staff-refpolicy-transplant

[root cause  ]: x
[changes     ]:x
[side effects]:x
[self test   ]:x
[download normally]:x
[power on/off normally]:x
[do common repository/branch inspection]:x
[is there dependence]:x
[confirm dependent commit]:x
[board]:x
[test case]:xx
[reviewers]:x

commit_template_version:v1
---
 policy/modules/roles/staff.if | 359 +++++++++++++++++++++++++++++++++-
 policy/modules/roles/staff.te |  44 +++++
 2 files changed, 402 insertions(+), 1 deletion(-)

diff --git a/policy/modules/roles/staff.if b/policy/modules/roles/staff.if
index 2bf7cbe1b..143072be3 100644
--- a/policy/modules/roles/staff.if
+++ b/policy/modules/roles/staff.if
@@ -153,7 +153,6 @@ interface(`staff_tmpfs_t_file',`
     allow $1 tmpfs_t:file {rw_file_perms};
 ')
 
-
 interface(`staff_pulseaudio_var_run_t_sock_file',`
     gen_require(`
         type pulseaudio_var_run_t;
@@ -162,3 +161,361 @@ interface(`staff_pulseaudio_var_run_t_sock_file',`
     allow $1 pulseaudio_var_run_t:sock_file {rw_sock_file_perms};
 ')
 
+interface(`staff_syslogd_t_unix_dgram_socket',`
+    gen_require(`
+        type syslogd_t;
+    ')
+
+    allow $1 syslogd_t:unix_dgram_socket { sendto };
+')
+
+interface(`staff_unreserved_port_t_tcp_socket',`
+    gen_require(`
+        type unreserved_port_t;
+    ')
+
+    allow $1 unreserved_port_t:tcp_socket { name_bind };
+')
+
+interface(`staff_event_device_t_chr_file',`
+	gen_require(`
+		type event_device_t;
+      ')
+
+    allow $1 event_device_t:chr_file {write};
+')
+
+interface(`staff_framebuf_device_t_chr_file',`
+	gen_require(`
+		type framebuf_device_t;
+      ')
+
+    allow $1 framebuf_device_t:chr_file {read write open ioctl map};
+')
+
+interface(`staff_fs_t_filesystem',`
+    gen_require(`
+		type fs_t;
+      ')
+
+    allow $1 fs_t:filesystem {remount};
+')
+
+interface(`staff_sysctl_fs_t_file',`
+    gen_require(`
+		type sysctl_fs_t;
+      ')
+
+     allow $1 sysctl_fs_t:file { read open};
+')
+
+################bug 1803864#########################
+interface(`staff_proc_t_dir',`
+     gen_require(`
+         type proc_t;
+     ')
+
+     allow $1 proc_t:dir { write add_name };
+')
+
+interface(`staff_proc_t_file',`
+		gen_require(`
+	         type proc_t;
+      ')
+
+      allow $1 proc_t:file { write create };
+')
+
+interface(`staff_sysctl_kernel_t_file',`
+      gen_require(`
+          type sysctl_kernel_t;
+      ')
+
+      allow $1 sysctl_kernel_t:file { write };
+')
+
+interface(`staff_sysctl_vm_t_dir',`
+    gen_require(`
+         type sysctl_vm_t;
+    ')
+
+    allow $1 sysctl_vm_t:dir { search };
+')
+
+interface(`staff_sysctl_vm_overcommit_t_file',`
+    gen_require(`
+          type sysctl_vm_overcommit_t;
+	')
+
+    allow $1 sysctl_vm_overcommit_t:file { read open };
+')
+
+interface(`staff_tmp_t_dir',`
+    gen_require(`
+          type tmp_t;
+        ')
+
+    allow $1 tmp_t:dir { setattr };
+')
+
+interface(`staff_sysfs_t_dir',`
+    gen_require(`
+        type sysfs_t;
+    ')
+    allow $1 sysfs_t:dir { write };
+')
+
+interface(`staff_dri_device_t_chr_file',`
+    gen_require(`
+        type dri_device_t;
+    ')
+    allow $1 dri_device_t:chr_file { setattr map };
+')
+
+#### BUG#1907204 ./factorytest reboot fail error ###
+interface(`staff_wtmp_t_file',`
+    gen_require(`
+        type wtmp_t;
+    ')
+    allow $1 wtmp_t:file { append };
+')
+
+interface(`staff_init_t_process',`
+    gen_require(`
+        type init_t;
+    ')
+    allow $1 init_t:process { signal };
+')
+
+#Bug 1921814
+interface(`staff_sysctl_t_file',`
+    gen_require(`
+        type sysctl_t;
+    ')
+    allow $1 sysctl_t:file { write open read };
+')
+
+interface(`staff_console_device_t_chr_file',`
+    gen_require(`
+        type console_device_t;
+    ')
+    allow $1 console_device_t:chr_file { write open ioctl };
+')
+
+interface(`staff_init_t_dbus',`
+    gen_require(`
+        class dbus send_msg;
+        type init_t;
+    ')
+
+    allow $1 init_t:dbus { send_msg };
+')
+
+interface(`staff_node_t_tcp_socket',`
+	gen_require(`
+		type node_t;
+	')
+	allow $1 node_t:tcp_socket { node_bind };
+')
+
+#1801603  WCN test permission denied#####
+
+interface(`staff_wireless_device_t_chr_file',`
+	gen_require(`
+		type wireless_device_t;
+        ')
+
+        allow $1 wireless_device_t:chr_file { write };
+')
+
+interface(`staff_commplex_link_port_t_tcp_socket',`
+	gen_require(`
+		type commplex_link_port_t;
+	')
+
+	allow $1 commplex_link_port_t:tcp_socket { name_bind };
+')
+
+#1803705  scp uploads permission denied#####
+
+interface(`staff_bin_t_file',`
+	gen_require(`
+		type bin_t;
+    ')
+
+    allow $1 bin_t:file { write create setattr };
+')
+
+interface(`staff_bin_t_dir',`
+	gen_require(`
+		type bin_t;
+    ')
+
+    allow $1 bin_t:dir { write add_name create };
+')
+
+interface(`staff_initctl_t_fifo_file',`
+    gen_require(`
+        type initctl_t;
+    ')
+
+    allow $1 initctl_t:fifo_file { write open };
+')
+
+interface(`staff_initrc_t_unix_stream_socket',`
+    gen_require(`
+        type initrc_t;
+    ')
+    allow $1 initrc_t:unix_stream_socket { connectto };
+')
+
+interface(`staff_staff_t_process',`
+    gen_require(`
+        type staff_t;
+    ')
+    allow $1 staff_t:process { execstack execmem };
+')
+
+interface(`staff_lib_t_file',`
+    gen_require(`
+        type lib_t;
+    ')
+    allow $1 lib_t:file { execmod };
+')
+
+##### Bug 1730001 unisoc sl8541e-smartpen ssh login function permission #####
+
+interface(`staff_removable_device_t_blk_file',`
+    gen_require(`
+        type removable_device_t;
+    ')
+
+    allow $1 removable_device_t:blk_file { write };
+')
+
+##### Bug 1731327 unisoc sl8541e-smartpen ssh login function permission #####
+interface(`staff_unlabeled_t_filesystem',`
+    gen_require(`
+        type unlabeled_t;
+    ')
+
+    allow $1 unlabeled_t:filesystem { getattr };
+')
+
+interface(`staff_dosfs_t_filesystem',`
+    gen_require(`
+        type dosfs_t;
+    ')
+
+    allow $1 dosfs_t:filesystem { unmount };
+')
+
+##### Bug 1805061 unisoc sl8541e-smartpen ssh tf card  permission #####
+interface(`staff_dosfs_t_dir',`
+        gen_require(`
+                type dosfs_t;
+        ')
+
+        allow $1 dosfs_t:dir { write add_name remove_name create rmdir rename };
+')
+
+interface(`staff_dosfs_t_file',`
+        gen_require(`
+                type dosfs_t;
+        ')
+
+        allow $1 dosfs_t:file { create write append unlink rename };
+')
+
+##### Bug 1814910 unisoc sl8541e-smartpen ssh exft tf card  permission #####
+interface(`staff_mnt_t_dir',`
+    gen_require(`
+        type mnt_t;
+    ')
+
+    allow $1 mnt_t:dir { write };
+')
+
+interface(`staff_staff_t_capability2',`
+    gen_require(`
+        type staff_t;
+    ')
+
+    allow $1 staff_t:capability2 { block_suspend };
+')
+
+interface(`staff_configfs_t_dir',`
+    gen_require(`
+        type configfs_t;
+    ')
+
+    allow $1 configfs_t:dir { read open };
+')
+
+interface(`staff_tracefs_t_dir',`
+    gen_require(`
+        type tracefs_t;
+    ')
+
+    allow $1 tracefs_t:dir { read open };
+')
+
+interface(`staff_sysctl_irq_t_dir',`
+    gen_require(`
+        type sysctl_irq_t;
+    ')
+
+    allow $1 sysctl_irq_t:dir { getattr read open };
+')
+
+interface(`staff_sysctl_net_unix_t_dir',`
+    gen_require(`
+        type sysctl_net_unix_t;
+    ')
+
+    allow $1 sysctl_net_unix_t:dir { getattr read open };
+')
+
+interface(`staff_fixed_disk_device_t_blk_file',`
+    gen_require(`
+        type fixed_disk_device_t;
+    ')
+
+    allow $1 fixed_disk_device_t:blk_file { read open ioctl };
+')
+
+##### Bug 1730617 unisoc sl8541e-smartpen ssh login function permission #####
+interface(`staff_staff_t_icmp_socket',`
+    gen_require(`
+        type staff_t;
+    ')
+
+    allow $1 staff_t:icmp_socket { create };
+')
+
+interface(`staff_staff_t_rawip_socket',`
+    gen_require(`
+        type staff_t;
+    ')
+
+    allow $1 staff_t:rawip_socket { create setopt getopt write read };
+')
+
+##### Bug 1741443 unisoc sl8541e-smartpen ssh login function permission #####
+interface(`staff_initrc_t_dbus',`
+    gen_require(`
+        class dbus send_msg;
+    ')
+
+    allow $1 initrc_t:dbus send_msg;
+')
+
+##### Bug 1793663 unisoc permission #####
+interface(`staff_kernel_t_process',`
+	gen_require(`
+		type kernel_t;
+      ')
+
+      allow $1 kernel_t:process {setsched};
+')
+
diff --git a/policy/modules/roles/staff.te b/policy/modules/roles/staff.te
index 05837d133..9b5db6fc5 100644
--- a/policy/modules/roles/staff.te
+++ b/policy/modules/roles/staff.te
@@ -233,4 +233,48 @@ staff_devlog_t_sock_file(staff_t)
 staff_device_t_chr_file(staff_t)
 staff_tmpfs_t_file(staff_t)
 staff_pulseaudio_var_run_t_sock_file(staff_t)
+staff_syslogd_t_unix_dgram_socket(staff_t)
+staff_unreserved_port_t_tcp_socket(staff_t)
+staff_event_device_t_chr_file(staff_t)
+staff_framebuf_device_t_chr_file(staff_t)
+staff_fs_t_filesystem(staff_t)
+staff_sysctl_fs_t_file(staff_t)
+staff_proc_t_dir(staff_t)
+staff_proc_t_file(staff_t)
+staff_sysctl_kernel_t_file(staff_t)
+staff_sysctl_vm_t_dir(staff_t)
+staff_sysctl_vm_overcommit_t_file(staff_t)
+staff_tmp_t_dir(staff_t)
+staff_sysfs_t_dir(staff_t)
+staff_dri_device_t_chr_file(staff_t)
+staff_wtmp_t_file(staff_t)
+staff_init_t_process(staff_t)
+staff_sysctl_t_file(staff_t)
+staff_console_device_t_chr_file(staff_t)
+staff_init_t_dbus(staff_t)
+staff_node_t_tcp_socket(staff_t)
+staff_wireless_device_t_chr_file(staff_t)
+staff_commplex_link_port_t_tcp_socket(staff_t)
+staff_bin_t_file(staff_t)
+staff_bin_t_dir(staff_t)
+staff_initctl_t_fifo_file(staff_t)
+staff_initrc_t_unix_stream_socket(staff_t)
+staff_staff_t_process(staff_t)
+staff_lib_t_file(staff_t)
+staff_removable_device_t_blk_file(staff_t)
+staff_unlabeled_t_filesystem(staff_t)
+staff_dosfs_t_filesystem(staff_t)
+staff_dosfs_t_dir(staff_t)
+staff_dosfs_t_file(staff_t)
+staff_mnt_t_dir(staff_t)
+staff_staff_t_capability2(staff_t)
+staff_configfs_t_dir(staff_t)
+staff_tracefs_t_dir(staff_t)
+staff_sysctl_irq_t_dir(staff_t)
+staff_sysctl_net_unix_t_dir(staff_t)
+staff_fixed_disk_device_t_blk_file(staff_t)
+staff_staff_t_icmp_socket(staff_t)
+staff_staff_t_rawip_socket(staff_t)
+staff_initrc_t_dbus(staff_t)
+staff_kernel_t_process(staff_t)
 
-- 
2.17.1

